### ISSUE TITLE
Align execution agent’s definition with the one in the standard

### ISSUE THREAD

**Issue by inbal2l** [executors/issaquah_2016#560]
From the wording in the paper, after invocation, the execution agent has reached the end of its lifetime. In case
of reallocating work to the thread (such as in thread pool) the different functionality to the same thread will be
considered as a newly allocated execution agent.

Section 3.2.5 in '[P2205R0: Executors Review - Polymorphic Executor](http://www.open-std.org/jtc1/sc22/wg21/docs/papers/2020/p2205r0.pdf)'

**Comment by jaredhoberock** [comment]
> From the wording in the paper, after invocation, the execution agent has reached the end of its lifetime. In case
of reallocating work to the thread (such as in thread pool) the different functionality to the same thread will be
considered as a newly allocated execution agent.

I think this is consistent with P0443's definition of execution agent. To take your example, a `static_thread_pool`'s threads aren't execution agents. Rather, an invocation of the function passed to `static_thread_pool::executor_type::execute` is an execution agent.

In other words, the `static_thread_pool`'s threads are the underlying resources on which execution agents run.

**Comment by inbal2l** [comment]
The problem is that the definition you've described above (resulting from P0443) is non consistent with the standard's current definition - more specifically:
```
32.2.5 Requirements for Cpp17Lockable types [thread.req.lockable]
32.2.5.1 In general [thread.req.lockable.general]
1 An execution agent is an entity such as a thread that may perform work in parallel with other execution
agents. [Note: Implementations or users can introduce other kinds of agents such as processes or thread-pool
tasks. — end note] The calling agent is determined by context, e.g., the calling thread that contains the
call, and so on.
 ```
(In addition to various other places, referring to execution agent as thread, for example: [coroutine.handle.resumption])
I've referred to the relevant section on P2205R0, I believe it describes the issue better.

**Comment by jaredhoberock** [comment]
Right. The idea is to supersede that definition with the one given in P0443, because the standard doesn't use it elsewhere
